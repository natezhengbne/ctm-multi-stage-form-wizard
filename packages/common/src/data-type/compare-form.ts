export type CompareForm = {
  personal?: Personal;
  address?: Address;
};

export type Personal = {
  firstName: string;
  lastName: string;
  email: string;
  phone?: string;
};

export type Address = {
  streetNumber: string;
  streetName: string;
  streetType: typeof streetTypes[number];
  suburb: string;
  postcode: string;
};

export const streetTypes = [
  "CI", "Ct", "St", "PI", "Ave",
]
