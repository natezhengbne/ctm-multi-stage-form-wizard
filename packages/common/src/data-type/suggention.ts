export type SuburbSuggestions = {
  _embedded: {
    suggestions: Suburb[];
  };
};

export type Suburb = {
  display: {
    text: string;
  };
  source: {
    postcode?: string;
    name?: string;
    state?: string;
  };
};
