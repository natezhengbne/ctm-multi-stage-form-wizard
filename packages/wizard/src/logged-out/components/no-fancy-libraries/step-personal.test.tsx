import { CompareForm } from "@ctm-multi-stage-wizard/common/data-type/compare-form";
import { shallow, ShallowWrapper } from "enzyme";
import React from "react";
import StepPersonal from "./step-personal";

describe("The page of step-personal renders", () => {
  it("renders correctly", () => {
    let wrapper = shallow(
      <StepPersonal step={1} activeStep={1} onNext={jest.fn()} onPrev={jest.fn()} value={undefined} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("page not active", () => {
    let wrapper = shallow(
      <StepPersonal step={1} activeStep={3} onNext={jest.fn()} onPrev={jest.fn()} value={undefined} />
    );
    expect(wrapper.html()).toEqual(null);
  });

  test("all form fields", () => {
    let wrapper = shallow(
      <StepPersonal step={1} activeStep={1} onNext={jest.fn()} onPrev={jest.fn()} value={undefined} />
    );

    expect(wrapper.find('input[id="firstName"]').props()).toEqual(
      expect.objectContaining({
        required: true,
        name: "firstName",
      })
    );
    expect(wrapper.find('input[id="lastName"]').props()).toEqual(
      expect.objectContaining({
        required: true,
        name: "lastName",
      })
    );
    expect(wrapper.find('input[id="email"]').props()).toEqual(
      expect.objectContaining({
        required: true,
        type: "email",
        name: "email",
      })
    );
    expect(wrapper.find('input[id="phone"]').props()).toEqual(
      expect.objectContaining({
        name: "phone",
      })
    );
    expect(wrapper.find('button[type="submit"]').length).toEqual(1);
  });
});

describe("The page of step-personal interaction", () => {
  let wrapper: ShallowWrapper;
  beforeEach(() => {
    wrapper = shallow(<StepPersonal step={1} activeStep={1} onNext={jest.fn()} onPrev={jest.fn()} value={undefined} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it("should set the first name value on change event", () => {
    wrapper.find('input[id="firstName"]').simulate("change", {
      target: {
        value: "testFirstName",
      },
    });
    expect(wrapper.find('input[id="firstName"]').prop("value")).toEqual("testFirstName");
  });

  it("should set the last name value on change event", () => {
    wrapper.find('input[id="lastName"]').simulate("change", {
      target: {
        value: "testLastName",
      },
    });
    expect(wrapper.find('input[id="lastName"]').prop("value")).toEqual("testLastName");
  });

  it("should set the email value on change event", () => {
    wrapper.find('input[id="email"]').simulate("change", {
      target: {
        value: "email@email.com",
      },
    });
    expect(wrapper.find('input[id="email"]').prop("value")).toEqual("email@email.com");
  });

  it("should set the phone value on change event", () => {
    wrapper.find('input[id="phone"]').simulate("change", {
      target: {
        value: "testPhone",
      },
    });
    expect(wrapper.find('input[id="phone"]').prop("value")).toEqual("testPhone");
  });
});

describe("The page of step-personal submit form", () => {
  let wrapper: ShallowWrapper;
  let nextCb: jest.Mock;
  let prevCb: jest.Mock;
  beforeEach(() => {
    nextCb = jest.fn((value: CompareForm) => {});
    prevCb = jest.fn();
    wrapper = shallow(<StepPersonal step={1} activeStep={1} onNext={nextCb} onPrev={prevCb} value={undefined} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test("subbmit form", () => {
    wrapper.find('input[id="firstName"]').simulate("change", {
      target: {
        value: "testFirstName",
      },
    });
    wrapper.find('input[id="lastName"]').simulate("change", {
      target: {
        value: "testLastName",
      },
    });
    wrapper.find('input[id="email"]').simulate("change", {
      target: {
        value: "email@email.com",
      },
    });
    wrapper.find('input[id="phone"]').simulate("change", {
      target: {
        value: "testPhone",
      },
    });
    wrapper.find('form[id="personal-form"]').simulate("submit", { preventDefault() {} });
    expect(nextCb).toBeCalledWith({
      personal: {
        firstName: "testFirstName",
        lastName: "testLastName",
        email: "email@email.com",
        phone: "testPhone",
      },
    } as CompareForm);
  });
});
