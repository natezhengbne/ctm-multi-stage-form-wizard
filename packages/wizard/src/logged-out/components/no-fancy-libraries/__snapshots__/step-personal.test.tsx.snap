// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The page of step-personal renders renders correctly 1`] = `
<Fragment>
  <form
    id="personal-form"
    onSubmit={[Function]}
  >
    <div
      className="StepperForm-root container-sm"
    >
      <h1>
        Personal
      </h1>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="firstName"
        >
          First Name *
        </label>
        <input
          className="form-control"
          id="firstName"
          maxLength={16}
          name="firstName"
          onChange={[Function]}
          required={true}
          type="text"
          value=""
        />
        <div
          className="invalid-feedback"
        >
          Please input your first name.
        </div>
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="lastName"
        >
          Last Name *
        </label>
        <input
          className="form-control"
          id="lastName"
          maxLength={16}
          name="lastName"
          onChange={[Function]}
          required={true}
          value=""
        />
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="email"
        >
          Email *
        </label>
        <input
          className="form-control"
          id="email"
          maxLength={20}
          name="email"
          onChange={[Function]}
          required={true}
          type="email"
          value=""
        />
      </div>
      <div
        className="mb-3"
      >
        <label
          className="form-label"
          htmlFor="phone"
        >
          Phone
        </label>
        <input
          className="form-control"
          id="phone"
          maxLength={20}
          name="phone"
          onChange={[Function]}
          pattern="^(\\\\+?\\\\(61\\\\)|\\\\(\\\\+?61\\\\)|\\\\+?61|\\\\(0[1-9]\\\\)|0[1-9])?( ?-?[0-9]){7,9}$"
          title="Please input valid phone number"
          value=""
        />
      </div>
      <div
        className="StepperActions-root"
      >
        <button
          className="btn btn-primary"
          type="submit"
        >
          Next
        </button>
      </div>
    </div>
  </form>
</Fragment>
`;
